### PR TITLE
add sdk-config 0.3 support for key/value headers and attributes

### PR DESCRIPTION
**Pull request by brettmc**
- headers and attributes are now arrays with `key` and `value` elements
- headers can optionally contain a `headers_list` element which supports a lower-priority CSV of key/value entries
- attributes can optionally contain an `attributes_list` element which supports a lower-priority CSV of key/value entries

Related: #1423

### CODE REVIEW

**Review by ChrisLightfootWild — COMMENTED** [review]
It's minor, but could we keep consistency and convert the snake-cased variables? 👼

File: src/Config/SDK/ComponentProvider/Logs/LogRecordExporterOtlp.php [hunk 1]
```diff
@@ -40,10 +41,13 @@ public function createPlugin(array $properties, Context $context): LogRecordExpo
     {
         $protocol = $properties['protocol'];
 
+        $headers_list = array_column(array_map(fn ($item) => explode('=', $item), explode(',', $properties['headers_list'] ?? '')), 1, 0);
+        $headers = array_column($properties['headers'], 'value', 'name') + $headers_list;
```
[the remark this answers is not in the document]
Comment (brettmc) [reply src/Config/SDK/ComponentProvider/Logs/LogRecordExporterOtlp.php]: We had an existing class which already does this, so I've updated to use MapParser

File: src/Config/SDK/ComponentProvider/Logs/LogRecordExporterOtlp.php [hunk 2]
```diff
@@ -31,7 +32,8 @@ final class LogRecordExporterOtlp implements ComponentProvider
      *     certificate: ?string,
      *     client_key: ?string,
      *     client_certificate: ?string,
-     *     headers: array<string, string>,
+     *     headers: array,
```
Comment (Nevay) [inline src/Config/SDK/ComponentProvider/Logs/LogRecordExporterOtlp.php:35]:
```suggestion
     *     headers: list<array{name: string, value: string}>,
```

**Comment by brettmc** [comment]
@open-telemetry/php-approvers I think this is ready for review now.

File: src/Config/SDK/ComponentProvider/Logs/LogRecordExporterOtlp.php [hunk 3]
```diff
@@ -31,7 +32,8 @@ final class LogRecordExporterOtlp implements ComponentProvider
      *     certificate: ?string,
      *     client_key: ?string,
      *     client_certificate: ?string,
-     *     headers: array<string, string>,
+     *     headers: list<array{name: string, value: string, type: ?string}>,
```
[the remark this answers is not in the document]
Comment (brettmc) [reply src/Config/SDK/ComponentProvider/Logs/LogRecordExporterOtlp.php]: yes, sorry. :(
fixed that.

**Review by ChrisLightfootWild — APPROVED** [review]

**Pull request merged**